### PR TITLE
expand the ATO checklist

### PR DESCRIPTION
**Pull request by afeld**
This pull request adds a bunch of items to the checklist, including new items to reflect what happens around and during the Sprint, as well as detailed tasks that should have been in there even before the Sprints were implemented. I recommend reviewing using the [rich diff](https://github.com/blog/1784-rendered-prose-diffs).

* Addresses parts of
    * https://github.com/18F/before-you-ship/issues/57
    * https://github.com/18F/before-you-ship/issues/191
    * https://github.com/18F/before-you-ship/issues/255
    * https://github.com/18F/before-you-ship/issues/259
* Closes https://github.com/18F/before-you-ship/issues/182.
* Closes https://github.com/18F/before-you-ship/issues/262.
* Closes https://github.com/18F/before-you-ship/issues/263.
* Closes https://github.com/18F/before-you-ship/issues/265.

### CODE REVIEW

**Comment by afeld** [comment]
Ready for 👀!

File: _includes/checklist.md [hunk 1]
```diff
@@ -1,41 +1,91 @@
...
+    - [ ] Add Project Plan template
+- [ ] Make a copy of the [notes doc template](https://docs.google.com/document/d/1EdcNyE1kkQve3tHyiV1QIRWNOBlTeh33lAbX0h4h18M/edit) in the [Sprinting Team folder](https://drive.google.com/open?id=1EdcNyE1kkQve3tHyiV1QIRWNOBlTeh33lAbX0h4h18M).
+    - [ ] Fill out the placeholders.
+    - [ ] Add link at the top of this issue.
+
+#### Project team
+
+##### Technical
+
+- [ ] Enable [protected branches](https://help.github.com/articles/about-protected-branches/) for the project repositor(ies).
```
Comment (afeld) [inline _includes/checklist.md]: Needs elaboration, probably in a follow-up pull request. https://github.com/18F/before-you-ship/issues/255#issuecomment-284143115

File: _includes/checklist.md [hunk 2]
```diff
@@ -1,41 +1,91 @@
...
 
 ## TODOs
 
-### Project team
+### ATO Sprint prerequisites
 
-Many of these tasks can be done in parallel. This is just a suggestion of priority.
+#### Infrastructure Lead
 
+- [ ] Set up an ATO intro meeting with the team.
```
Comment (abisker) [inline _includes/checklist.md]: Can we update it to say 'project team'?

File: _includes/checklist.md [hunk 3]
```diff
@@ -1,41 +1,91 @@
...
+- [ ] Determine the [impact level](https://pages.18f.gov/before-you-ship/ato/levels/).
```
Comment (abisker) [inline _includes/checklist.md]: Does this need to be confirmed by anyone? And once determined, do you want it listed here and/or where should it be recorded?

File: _includes/checklist.md [hunk 4]
```diff
@@ -1,41 +1,91 @@
...
+- [ ] Add this issue to the `Backlog` of [the ATO Kanban board](https://github.com/18F/Infrastructure/projects/1).
+- [ ] Assign the appropriate labels to this issue.
```
Comment (abisker) [inline _includes/checklist.md]: Suggest an indent here

File: _includes/checklist.md [hunk 5]
```diff
@@ -1,41 +1,91 @@
...
+- [ ] Set up the project ATO folder.
+    - [ ] Create as a sub-folder of [the `ATOs` folder in Google Drive](https://drive.google.com/a/gsa.gov/folderview?id=0BynIxtx-CfkdckljM3BPSkdQT1U&usp=sharing) under 18F/OPP/PIF then "Work in progress". Link to it above.
```
Comment (abisker) [inline _includes/checklist.md]: Can we give guidance about title? To help make them as uniform as possible? i.e., is it "90 Day CALC ATO"  or "CALC ATO - 90 Day" or...?

File: _includes/checklist.md [hunk 6]
```diff
@@ -1,41 +1,91 @@
...
+    - [ ] Add Rules of Engagement (RoE) template
```
Comment (abisker) [inline _includes/checklist.md]: is there a link for this one?

File: _includes/checklist.md [hunk 7]
```diff
@@ -1,41 +1,91 @@
...
+    - [ ] Add [System Security Plan (SSP)](https://pages.18f.gov/before-you-ship/ato/ssp/) template
+    - [ ] Add Project Plan template
```
Comment (abisker) [inline _includes/checklist.md]: is there a link for this one?

File: _includes/checklist.md [hunk 8]
```diff
@@ -1,41 +1,91 @@
...
+- [ ] Make a copy of the [notes doc template](https://docs.google.com/document/d/1EdcNyE1kkQve3tHyiV1QIRWNOBlTeh33lAbX0h4h18M/edit) in the [Sprinting Team folder](https://drive.google.com/open?id=1EdcNyE1kkQve3tHyiV1QIRWNOBlTeh33lAbX0h4h18M).
```
Comment (abisker) [inline _includes/checklist.md]: This is slightly confusing coming right after the template bullets above... would suggest updating to something like "Make a copy of the 'notes doc template' and save it inside of the Sprinting team folder, titling it 'Naming Convention' " 

Also, may want to make the 'notes doc template' title more descriptive/precise.

File: _includes/checklist.md [hunk 9]
```diff
@@ -1,41 +1,91 @@
...
+    - [ ] Fill out the placeholders.
+    - [ ] Add link at the top of this issue.
```
Comment (abisker) [inline _includes/checklist.md]: Add the document link to the "Sprint Notes" section at the top of this issue

File: _includes/checklist.md [hunk 10]
```diff
@@ -1,41 +1,91 @@
...
+
+#### Project team
+
+##### Technical
+
+- [ ] Enable [protected branches](https://help.github.com/articles/about-protected-branches/) for the project repositor(ies).
+    * Get admin access via #admins-github, if needed.
+- [ ] Ensure that a staging environment is fully set up.
+    * This should be as [production](https://pages.18f.gov/before-you-ship/infrastructure/)-like as possible.
 - [ ] [Set up monitoring](https://pages.18f.gov/before-you-ship/infrastructure/monitoring/)
```
Comment (abisker) [inline _includes/checklist.md]: I'm unclear if you want links to these (or some proof that this has been set up) linked to in the issue?

File: _includes/checklist.md [hunk 11]
```diff
@@ -1,41 +1,91 @@
...
     * [ ] [Downtime alerts](https://pages.18f.gov/before-you-ship/infrastructure/monitoring/#downtime)
-    * [ ] [Error alerts](https://pages.18f.gov/before-you-ship/infrastructure/monitoring/#errors)
-- [ ] Add an [`.about.yml`](https://github.com/18F/about_yml) for the main repository
-- [ ] Create a sub-folder in [the `ATOs` folder in Google Drive](https://drive.google.com/a/gsa.gov/folderview?id=0BynIxtx-CfkdckljM3BPSkdQT1U&usp=sharing) under 18F/OPP/PIF then "Work in progress". This will be knows as the "ATO Folder". Link it here.
-- [ ] Security scans (make this a link to the ATO Folder where you put a copy of the security scan results)
+    * [ ] [Error & performance alerts](https://pages.18f.gov/before-you-ship/infrastructure/monitoring/#errors--performance-problems)
+- [ ] Perform security scans, and put the results (or a link to them) in the ATO folder.
     * [ ] Set up [static analysis](https://pages.18f.gov/before-you-ship/security/static-analysis/) service
```
Comment (abisker) [inline _includes/checklist.md]: Same question here, I'm unclear if you want links to this (or some proof that this has been set up) linked to in the issue?

File: _includes/checklist.md [hunk 12]
```diff
@@ -1,41 +1,91 @@
...
         * [ ] Add service badges to the README
+        * [ ] Put a point-in-time PDF of the scan results in the ATO folder
     * [ ] [Perform dynamic vulnerability scanning](https://pages.18f.gov/before-you-ship/security/dynamic-scanning/)
         * [ ] Resolve any visible security issues, re-running the scan as needed
         * [ ] Add the issue-free scan report or [documentation about false positives](https://pages.18f.gov/before-you-ship/security/dynamic-scanning/#caveats) to the ATO folder.
```
Comment (abisker) [inline _includes/checklist.md]: small naming thing, but when I read 'the ATO folder' I think the top level one... when it says 'your ATO folder' I think of the correct place

File: _includes/checklist.md [hunk 13]
```diff
@@ -1,41 +1,91 @@
...
+- [ ] If this is a new system, add a prominent [`Beta`](https://18f.gsa.gov/dashboard/stages/) label to the site
```
Comment (abisker) [inline _includes/checklist.md]: to the production site or staging site?

File: _includes/checklist.md [hunk 14]
```diff
@@ -1,41 +1,91 @@
...
+- [ ] Ensure the staging environment has sufficient capacity to withstand the testing
```
Comment (abisker) [inline _includes/checklist.md]: should we add a note about how to adjust/get more? refer to cloud.gov team?

File: _includes/checklist.md [hunk 15]
```diff
@@ -1,41 +1,91 @@
...
+    * The testing tools create very heavy usage and traffic.
+
+##### Content
```
Comment (abisker) [inline _includes/checklist.md]: Maybe a more specific title? "Prep Work and Document Content (or Document Population)"

File: _includes/checklist.md [hunk 16]
```diff
@@ -1,41 +1,91 @@
...
+* **Product manager:** @[username]
+* **System Owner:** @[username of technical point of contact who will be the project representative in the Sprint]
+* **Infrastructure Lead:** @[username]
+* **ATO folder:** [url]
+* **Sprint notes:** [url]
 
 ## TODOs
 
-### Project team
+### ATO Sprint prerequisites
```
Comment (abisker) [inline _includes/checklist.md]: Can we add a descriptor here along the lines of "These three checklists (Infra Lead and Project Team Technical and Project Team Content) must be complete before you may enter a sprint"

File: _includes/checklist.md [hunk 17]
```diff
@@ -1,41 +1,91 @@
...
-- [ ] Add a System Security Plan Google doc to the ATO folder (talk to your Infrastructure Lead for a template)
+- [ ] Add a [System Security Plan YAML file](https://pages.18f.gov/before-you-ship/ato/ssp/#template) to the repository
 - [ ] [Set up Compliance Masonry documentation](https://github.com/18F/cg-compliance#starting-ato-documentation-for-cloudgov-applications)
-- [ ] [Implement the controls](https://pages.18f.gov/before-you-ship/ato/walkthrough/#step-3--implement-the-controls)
-- [ ] Add a Rules of Engagement (RoE) Google doc to the ATO folder (talk to your Infrastructure Lead for a template)
-- [ ] Ensure system environment referenced in the RoE for security testing is ready for testing. No production links should be included in the RoE so as to avoid any testing happening on production (which could lead to downtime).
-- [ ] Add a Project Plan Google doc to the ATO folder (talk to your Infrastructure Lead for a template)
 
-### Authorizing Official
+### Documentation review
```
Comment (abisker) [inline _includes/checklist.md]: It's not clear that this is a certain point in the process/in time. Maybe a sentence between the above and this outlining that now you are entering the sprint process (and a link describing the process)?

File: _includes/checklist.md [hunk 18]
```diff
@@ -1,41 +1,91 @@
...
+
+1. [ ] Move this issue to the `Documentation review` column of [the ATO Kanban board](https://github.com/18F/Infrastructure/projects/1). - @[infrastructure lead]
+1. [ ] Schedule a documentation review session. - @[infrastructure lead]
+    * One or more follow-up sessions may be necessary.
+1. [ ] Fix any documentation issues identified in the session.
+1. [ ] RoE signed
+    * [ ] System Owner
+    * [ ] GSA IT
+
+### Mid-Sprint
```
Comment (abisker) [inline _includes/checklist.md]: It seems odd to jump from doc review to mid-sprint, shouldn't it be start of sprint?

File: _includes/checklist.md [hunk 19]
```diff
@@ -1,41 +1,91 @@
...
+
+1. [ ] Penetration test complete. - @[tester]
```
Comment (abisker) [inline _includes/checklist.md]: Can we add a bullet for initial meeting with pen test team and/or something about it beginning? so the first bullet isn't it being complete?

File: _includes/checklist.md [hunk 20]
```diff
@@ -1,41 +1,91 @@
...
+    * [ ] Enhanced Scanning and Assessment Process (ESAP) document added to ATO folder
```
Comment (abisker) [inline _includes/checklist.md]: Add something about who creates this

File: _includes/checklist.md [hunk 21]
```diff
@@ -1,41 +1,91 @@
...
+1. [ ] Put all findings in project's issue tracker.
```
Comment (abisker) [inline _includes/checklist.md]: Findings of what? The Pentest?

File: _includes/checklist.md [hunk 22]
```diff
@@ -1,41 +1,91 @@
...
+1. [ ] Fix any `Critical` or `High` vulnerabilities.
```
Comment (abisker) [inline _includes/checklist.md]: "...as identified by the pentest/listed in the ESAP"

File: _includes/checklist.md [hunk 23]
```diff
@@ -1,41 +1,91 @@
...
+    * This needs to be done before the ATO can be issued, but not necessarily before the end of the sprint.
+
+### Post-Sprint
 
-* [ ] Final review and risk acceptance signatures (ATO)
+1. [ ] [Controls](https://pages.18f.gov/before-you-ship/ato/controls/) tested - @[GSA IT representative]
+1. [ ] Create a Plan of Actions and Milestones (POAM). - @[GSA IT representative]
+1. [ ] Final review and risk acceptance signatures (issue the ATO) - @NoahKunin
+1. [ ] Remove the `Beta` label from the site.
```
Comment (abisker) [inline _includes/checklist.md]: production or staging? or both?

File: _includes/checklist.md [hunk 24]
```diff
@@ -1,41 +1,91 @@
...
+1. [ ] Fix all `Moderate` vulnerabilities - due [30 days after ATO issued]
```
Comment (abisker) [inline _includes/checklist.md]: "...as identified by the pentest/listed in the ESAP"

File: _includes/checklist.md [hunk 25]
```diff
@@ -1,41 +1,91 @@
...
+1. [ ] Fix all `Low` vulnerabilities - due [60 days after ATO issued]
```
Comment (abisker) [inline _includes/checklist.md]: "...as identified by the pentest/listed in the ESAP"

File: _includes/checklist.md (hunk 3 above)
Comment (hollyallen) [inline _includes/checklist.md]: It should be agreed upon with the AO. Good idea to list it here.

File: _pages/ato/checklist.md [hunk 26]
```diff
@@ -3,7 +3,9 @@ title: ATO Checklist
 navtitle: Checklist
 ---
 
-To start the security clearance process, [create an issue in the Infrastructure repository](https://github.com/18F/Infrastructure/issues/new?title=ATO+for+%3Cproject%3E+-+due+%3Cdeadline%3E) using [this template](https://raw.githubusercontent.com/18F/before-you-ship/master/_includes/checklist.md). Make sure to replace the placeholders at the top. Feel free to add a username after each task to assign it, and/or make corresponding items in your issue tracker. After the Infrastructure issue is created, the Infrastructure Team will schedule a time to meet with you and discuss the ATO.
+To start the security clearance process, [create an issue in the Infrastructure repository](https://github.com/18F/Infrastructure/issues/new?title=ATO+for+%5Bproject%5D-+due+%5Bdeadline%5D) using [this template](https://raw.githubusercontent.com/18F/before-you-ship/master/_includes/checklist.md) as the body. Make sure to replace the placeholders (the things in `[square brackets]`). Feel free to add a username after each task to assign it, and/or make corresponding items in your issue tracker. Unless otherwise specified, all tasks are the responsibility of the project team.
```
Comment (abisker) [inline _pages/ato/checklist.md]: Does this appear at the top? 

Need more clarity on replacing the placeholders throughout.

File: _includes/checklist.md (hunk 5 above)
Comment (hollyallen) [inline _includes/checklist.md]: Good idea. Personally I find "<project name> ATO - <duration> <level>" to be easy to read.

File: _includes/checklist.md (hunk 6 above)
Comment (hollyallen) [inline _includes/checklist.md]: https://insite.gsa.gov/portal/content/627238 even though the template is called "Rules of Engagement (RoE) 90-Day LATO Penetration Test TEMPLATE" it is for use with all engagements.

File: _includes/checklist.md (hunk 7 above)
Comment (hollyallen) [inline _includes/checklist.md]: Sadly no, we are just using the last one we made. This is a good thing to work on, as much of the project plan doesn't feel applicable to the LATO process.

File: _includes/checklist.md (hunk 8 above)
Comment (hollyallen) [inline _includes/checklist.md]: Could add "ato sprinting" to the words here (it is already part of the google doc file).

File: _includes/checklist.md (hunk 10 above)
Comment (hollyallen) [inline _includes/checklist.md]: The link should be at the top of the issue, as the site url.

File: _includes/checklist.md (hunk 23 above)
Comment (hollyallen) [inline _includes/checklist.md]: both

File: _includes/checklist.md (hunk 14 above)
Comment (hollyallen) [inline _includes/checklist.md]: Disagree, that level of technical detail would be inappropriate for this checklist. Dev teams know what to do.

File: _includes/checklist.md (hunk 13 above)
Comment (hollyallen) [inline _includes/checklist.md]: Both

File: _includes/checklist.md (hunk 10 above)
Comment (hollyallen) [inline _includes/checklist.md]: No need for proof for these things. The link to the staging/testing environment will be at the top of the list already.

File: _includes/checklist.md (hunk 4 above)
Comment (afeld) [inline _includes/checklist.md]: _Technically_ they are independent tasks. Cool if I leave it this way?

Comment (abisker) [inline _includes/checklist.md]: yep

File: _pages/ato/checklist.md (hunk 26 above)
Comment (afeld) [inline _pages/ato/checklist.md]: This page appears in Before You Ship, but not in the checklist template (and thus, not the ATO issue that will be created).

File: _includes/checklist.md (hunk 13 above)
Comment (afeld) [inline _includes/checklist.md]: Would saying "from all live sites" be confusing? Theoretically, they change it in the codebase and it shows up everywhere.

**Comment by afeld** [comment]
Updated!

**Pull request merged**